Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -22,7 +22,26 @@ namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
-  public class Program {
+  public interface IHasUsages : IDeclarationOrUsage {
```

Comment:
Could you please move IHasUsage after IDeclarationOrUsage, and IDeclarationOrUsage after Node? It makes more sense to me to describe something linearly if possible, starting at the top ancestors